control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -756,14 +861,17 @@ func (r *AWSEndpointServiceReconciler) createSecurityGroup(ctx context.Context,
 	describeSGInput := &ec2.DescribeSecurityGroupsInput{
 		GroupIds: []*string{aws.String(sgID)},
 	}
-	if err = r.ec2Client.WaitUntilSecurityGroupExistsWithContext(ctx, describeSGInput); err != nil {
+	if err = ec2Client.WaitUntilSecurityGroupExistsWithContext(ctx, describeSGInput); err != nil {
+		log.Error(err, "failed to wait for security group to exist", "id", sgID)
```

Comment:
why are we logging in addition to the returned error and with different message?

Reply:
The difference is that the returned error only contains the error code (it goes in the status of the resource) while the log line includes the full error returned from AWS.

I fixed the different error messages

Reply:
So for example here we'll log 
"failed to wait for security group to exist", "id", sgID"
and the controller runtime will log
"failed to wait for security group to exist (id: %s), code: %s", sgID, supportawsutil.AWSErrorCode(err)"

Seems unncecesary?

Reply:
The difference is that the latter will also go into the status of the AWSEndpointService